tests/sqlalchemy/test_patch.py

```diff
@@ -48,18 +48,36 @@ def test_inspector_patched(database, needs_sqlalchemy2):
 def test_json_encoder_date():
     """
     Verify the extended JSON encoder also accepts Python's `date` types.
+
+    TODO: Move to different test file, as this no longer requires
+          monkeypatching after using orjson for marshalling.
     """
     data = {"date": datetime.date(2024, 6, 4)}
-    encoded = json.dumps(data, cls=CrateJsonEncoderWithNumPy)
-    assert encoded == '{"date": 1717459200000}'
+    # TODO: Refactor core marshalling method out of crate-python's
+    #       `_create_sql_payload`, to make it reusable here.
+    encoded = orjson.dumps(
+        data,
+        default=cratedb_json_encoder,
+        option=(orjson.OPT_PASSTHROUGH_DATETIME | orjson.OPT_SERIALIZE_NUMPY),
+    )
```

Comment:
- See https://github.com/crate/crate-python/pull/691#pullrequestreview-2555613789.

Reply:
This has been broken out into a canonical `json_dumps` function, so code duplication is no longer necessary.